modules/cxf/src/main/java/org/mule/module/cxf/CxfInboundMessageProcessor.java

```diff
@@ -229,150 +235,192 @@ protected MuleEvent sendToDestination(MuleEvent event) throws MuleException, IOE
     {
         try
         {
-            final MessageImpl m = new MessageImpl();
-            final MuleMessage muleReqMsg = event.getMessage();
-            String method = muleReqMsg.getInboundProperty(HttpConnector.HTTP_METHOD_PROPERTY);
+            final Exchange exchange = new ExchangeImpl();
 
-            String ct = muleReqMsg.getInboundProperty(HttpConstants.HEADER_CONTENT_TYPE);
-            if (ct != null)
+            if (event.isAllowNonBlocking())
             {
-                m.put(Message.CONTENT_TYPE, ct);
+                final ReplyToHandler originalReplyToHandler = event.getReplyToHandler();
+
+                event = new DefaultMuleEvent(event, new ReplyToHandler()
+                {
+                    @Override
+                    public void processReplyTo(MuleEvent event, MuleMessage returnMessage, Object replyTo) throws MuleException
+                    {
+                        try
+                        {
+                            originalReplyToHandler.processReplyTo(event, returnMessage, replyTo);
```

Comment:
It's not being created. The output handler expects a DelegatingOutputStream to be the content of the CXF output message received in the exchange. In the NB case, we already have the returning mule event, which have no output stream set.